tests/hipscat/pixel_math/test_margin_bounding.py

```diff
@@ -0,0 +1,299 @@
+    # far out point, expect False
+    test_ra1 = 90.
+    test_dec1 = -2.
+
+    # middle of our healpixel, expect True
+    test_ra2 = -100.
+    test_dec2 = -45.
+
+    test_ra = np.array([test_ra1, test_ra2])
+    test_dec = np.array([test_dec1, test_dec2])
+
+    test_sc = SkyCoord(test_ra, test_dec, unit="deg")
+
+    scale = pm.get_margin_scale(0, 0.1)
+    bounds = pm.get_margin_bounds_and_wcs(0, 5, scale)
+
+    assert len(bounds) == 16
+
+    vals = []
+    for p, w in bounds:
+        x, y = w.world_to_pixel(test_sc)
+
+        pc = PixCoord(x, y)
+        vals.append(p.contains(pc))
+
+    checks = np.array(vals).any(axis=0)
+    
+    expected = np.array([True, False])
+
+    npt.assert_array_equal(checks, expected)
+
+def test_get_margin_bounds_and_wcs_north_pole():
+    """Make sure get_margin_bounds_and_wcs works at pixels along the north polar region"""
+    scale = pm.get_margin_scale(1, 0.1)
+    bounds = pm.get_margin_bounds_and_wcs(1, 7, scale, step = 100)
+
+    assert len(bounds) == 4
+
+    test_ra = np.array([50, 100., 130., 130., 100.])
+    test_dec = np.array([-60., 89.9, 65., 85., 50.])
+
+    test_sc = SkyCoord(test_ra, test_dec, unit="deg")
+
+    vals = []
+    for p, w in bounds:
+        x, y = w.world_to_pixel(test_sc)
+
+        pc = PixCoord(x, y)
+        vals.append(p.contains(pc))
+
+    checks = np.array(vals).any(axis=0)
+    
+    expected = np.array([False, True, True, True, False])
+
+    npt.assert_array_equal(checks, expected)
+
+def test_get_margin_bounds_and_wcs_south_pole():
+    """Make sure get_margin_bounds_and_wcs works at pixels along the south polar region"""
+    scale = pm.get_margin_scale(1, 0.1)
+    bounds = pm.get_margin_bounds_and_wcs(1, 36, scale)
+
+    assert len(bounds) == 4
+
+    test_ra = np.array([50, 120., 108., 150., 100., 104.])
+    test_dec = np.array([-60., -70., -66.2, -70., -90., -80])
+
+    test_sc = SkyCoord(test_ra, test_dec, unit="deg")
+
+    vals = []
+    for p, w in bounds:
+        x, y = w.world_to_pixel(test_sc)
+
+        pc = PixCoord(x, y)
+        vals.append(p.contains(pc))
+
+    checks = np.array(vals).any(axis=0)
+    
+    expected = np.array([False, True, True, True, False, True])
+
+    npt.assert_array_equal(checks, expected)
+
+def test_get_margin_bounds_and_wcs_ra_rollover():
+    """Make sure get_margin_bounds_and_wcs works at the rollover point for right ascension"""
+    scale = pm.get_margin_scale(1, 0.1)
+    bounds = pm.get_margin_bounds_and_wcs(1, 27, scale)
+
+    assert len(bounds) == 4
+
+    test_ra = np.array([180., 0.])
+    test_dec = np.array([20., 0.])
+
+    test_sc = SkyCoord(test_ra, test_dec, unit="deg")
+
+    vals = []
+    for p, w in bounds:
+        x, y = w.world_to_pixel(test_sc)
+
+        pc = PixCoord(x, y)
+        vals.append(p.contains(pc))
+
+    checks = np.array(vals).any(axis=0)
+    
+    expected = np.array([True, False])
+
+    npt.assert_array_equal(checks, expected)
```

Comment:
you were right that the computed centroid was incorrect and causing a lot of issues. the solution was to normalize points of pixels around the ra 180 -> -180 rollover point to a 0 -> 360 scale so that all the math works properly. I went ahead and added the given values to the test as well.

also in the process of fixing this bug I discovered a bug when checking points that are exactly the border of a border region slice where they were showing up as `False` when bound checking. I extended the number of boundary points in a slice so that there is now overlap between the polygons and not just a hard border 🙂